lte/gateway/c/session_manager/UpfMsgManageHandler.cpp

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright 2020 The Magma Authors.
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree.
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include "UpfMsgManageHandler.h"
+#include <google/protobuf/util/time_util.h>
+#include <chrono>
+#include <thread>
+#include "magma_logging.h"
+#include "GrpcMagmaUtils.h"
+#include "lte/protos/session_manager.pb.h"
+
+using grpc::Status;
+
+namespace magma {
+/**
+ * SetInterfaceForUserPlaneHandler processes gRPC requests for the sessionD
+ * This composites the all the request that comes from UPF
+ */
+
+UpfMsgManageHandler::UpfMsgManageHandler(
+    std::shared_ptr<SessionStateEnforcer> enforcer, SessionStore& session_store)
+    : session_store_(session_store), conv_enforcer_(enforcer) {}
+
+/**
+ * Node level GRPC message received from UPF
+ * during startup
+ */
+
+void UpfMsgManageHandler::SetUPFNodeState(
+    ServerContext* context, const UPFNodeState* node_request,
+    std::function<void(Status, SmContextVoid)> response_callback) {
+  auto& request = *node_request;
+  // Print the message from UPF
+  PrintGrpcMessage(static_cast<const google::protobuf::Message&>(request));
+  MLOG(MDEBUG) << "Node UPF details :";
+  conv_enforcer_->get_event_base().runInEventBaseThread([this, request]() {
+    UPFNodeState::UpfNodeMessagesCase msgtype =
+        request.upf_node_messages_case();
+    if (msgtype == UPFNodeState::kAssociatonState) {
```

Comment:
This is more preference but this looks like it could be nicer as a switch statement.
```
switch (request.upf_node_messages_case()) {
	case UPFNodeState::kAssociatonState:
		// logic		
		break;
	default:
		// log something about the message type not being handled
}
```